#BN-34 chores: setup NodeJs rest api

EvelyneUmubyeyi opened the pull request:
#### What does this PR do?
In this PR I set up the project structure to help start building RESTful web APIs using Express framework in Node Js.

#### Description of Task to be completed
I installed the primary dependencies like express, dotenv, and so forth. I also set up the swagger documentation by installing the necessary dependencies, and adding the necessary codes. Further, I added ESLint setup to ensure code quality.

#### How should this be manually tested?
You can run the project using npm start which will start the server hitting the welcome endpoint. You can also try http://localhost:3000/api-docs/ for the documentation.

UmubyeyiEvelyne committed:
Added node js API project setup

UmubyeyiEvelyne committed:
Added new folders in the src folder namely controllers, helpers, middleware, services and utils

UmubyeyiEvelyne committed:
Added description in the index files of each folder in the src folder to describe what each folder will contain

chrisostome01 commented on src/docs/apidoc.js:
```diff
@@ -0,0 +1,29 @@
+    },
+  },
+  servers: [
+    {
+      url: 'http://localhost:3000/',
+      description: 'Api server',
+    },
```
Since this link is dynamic according to deployments, would you keep it in the env file and have the URL conditional. if the environment variable of the URL exists assign it if it does not keep the localhost with that port.

chrisostome01 commented on src/docs/apidoc.js:
```diff
@@ -0,0 +1,29 @@
+  definition: {
+    openapi: '3.0.0',
+    info: {
+      title: 'API Library',
+      version: 1.0,
+      description: 'Swagger Api Documentation',
```
It would be better if this description would relate to the project

chrisostome01 reviewed (changes requested):
# Well done @EvelyneUmubyeyi 
I left some comments for the improvement and future use case, please check them out.
Otherwise, this was a such a good setup.

UmubyeyiEvelyne committed:
chore(#BN-34: setup NodeJs rest api): I created a pull request template and modified the swagger docummentation options

- create the PULL_REQUEST_TEMPLATE.md inside .github folder in the home directory
- I changed the server url used in swagger options from being hardcoded to a variable stored in the .env file and added a more detailed description
